src/parsers/Uniref_mapping.py

```diff
@@ -0,0 +1,467 @@
+            "members": members,
+            "created": now,
+            "cdm_version": "1.0",
+        }
+
+        cdm_results.append(cdm)
+
+    return cdm_results
+
+
+def flatten_cdm_entry(cdm):
+    """
+    Flat CDM structure, one row per member, retain clustering attributes + all attributes for that member
+    """
+    basic = {
+        "entity_id": cdm["entity_id"],
+        "entity_type": cdm["entity_type"],
+        "source": cdm["source"],
+        "uniref_level": cdm["uniref_level"],
+        "cluster_id": cdm["cluster_id"],
+        "updated": cdm["updated"],
+        "member_count": cdm["member_count"],
+        "functional_annotation": cdm["functional_annotation"],
+        "common_taxon": cdm["common_taxon"],
+        "common_taxon_id": cdm["common_taxon_id"],
+        "created": cdm["created"],
+        "cdm_version": cdm["cdm_version"],
+    }
+
+    # Represents a member attribute, prefixed with rep_ to avoid conflicts with member fields
+    rep = cdm.get("representative", {})
+    rep_flat = {f"rep_{k}": v for k, v in rep.items()}
+
+    records = []
+    # Iterate over all members, output one line per member
+    for i in cdm.get("members", []):
+        ## Add mem_ prefix to each member field
+        mem_flat = {f"mem_{k}": v for k, v in i.items()}
+        ## merge all fields into one row
+        row = {**basic, **rep_flat, **mem_flat}
+        records.append(row)
+
+    if not records:
+        row = {**basic, **rep_flat}
+        records.append(row)
+    return records
+
+
+def main():
+    tsv_path = "uniref100_ids_part.tsv"
+    max_count = 1000
+    page_size = 100
+
+    ## Get a list of UniRef100 group IDs
+    if not os.path.exists(tsv_path):
+        fetch_uniref100_ids(out_tsv=tsv_path, max_ids=max_count, page_size=page_size)
+
+    ## Read the list of IDs and download all XMLs
+    with open(tsv_path) as fin:
+        ids = [
+            line.strip()
+            for line in fin
+            if line.strip() and not line.startswith("Cluster ID")
+        ]
+    batch_download_uniref_xml(ids)
+
+    ## Parses all XML to CDM
+    all_cdm_entries = []
+    xml_files = glob.glob("UniRef100_*.xml")
+
+    for xml_file in xml_files:
+        try:
+            all_cdm_entries.extend(parse_uniref_xml_to_cdm(xml_file))
+            print(f"Parsed {xml_file}")
+        except Exception as e:
+            print(f"Error parsing {xml_file}: {e}")
+
+    ## Export as jsonl
+    with open("all_uniref100_cdm.jsonl", "w") as fout:
+        for cdm in all_cdm_entries:
+            fout.write(json.dumps(cdm, ensure_ascii=False) + "\n")
+    print("All clusters written to all_uniref100_cdm.jsonl")
+
+    ## Spread export to csv and parquet
+    all_flat = []
+    for cdm in all_cdm_entries:
+        all_flat.extend(flatten_cdm_entry(cdm))
+
+    df_flat = pd.DataFrame(all_flat)
+    df_flat.to_csv("all_uniref100_flat.csv", index=False)
+    df_flat.to_parquet("all_uniref100_flat.parquet", index=False)
```

Comment:
why are there all these different export formats?

Reply:
I thought that was why but just checking -- hopefully things will be clearer later on! 🙂